src/lib/rnp2.c

```diff
@@ -2075,8 +2077,30 @@ rnp_op_verify_signature_get_hash(rnp_op_verify_signature_t sig, char **hash)
 rnp_result_t
 rnp_op_verify_signature_get_key(rnp_op_verify_signature_t sig, rnp_key_handle_t *key)
 {
-    // TODO : Implement this
-    return RNP_ERROR_NOT_IMPLEMENTED;
+    rnp_ffi_t ffi = sig->ffi;
+    pgp_key_search_t search;
+
+    // create a search (since we'll use this later anyways)
+    search.type = PGP_KEY_SEARCH_KEYID;
+    memcpy(search.by.keyid, sig->keyid, PGP_KEY_ID_SIZE);
+
+    // search the stores
+    pgp_key_t *pub = rnp_key_store_search(&ffi->io, ffi->pubring, &search, NULL);
+    pgp_key_t *sec = rnp_key_store_search(&ffi->io, ffi->secring, &search, NULL);
+    if (!pub && !sec) {
```

Comment:
My thoughts were that it's just a general "key handle", so it's not necessarily about what verification requires, it's about presenting a consistent interface to clients.
For example, it seems like it might be weird to get a key handle with `rnp_locate_key` which has true values for both `rnp_key_have_public`/`rnp_key_have_secret`, but the same key handle from the function above would be different if we didn't fill in that sec pointer.

I guess it depends on how you view a key handle though.

I've been thinking about alternative approaches to the key handle for a while now, so I'll create a GH issue to discuss that.